robot/impl/local_robot.go

```diff
@@ -402,19 +402,29 @@ func newWithResources(
 	}()
 
 	if cfg.Cloud != nil && cfg.Cloud.AppAddress != "" {
-		_, cloudConn, err := r.cloudConnSvc.AcquireConnection(ctx)
-		if err == nil {
-			r.packageManager, err = packages.NewCloudManager(cfg.Cloud, pb.NewPackageServiceClient(cloudConn), cfg.PackagePath, logger)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			if !errors.Is(err, context.DeadlineExceeded) {
-				return nil, err
+		connectionChan := make(chan packages.DeferredConnectionResponse)
+		goutils.PanicCapturingGo(func() {
+			for {
+				// deferred connection manager uses this for-loop behavior to
+				// implement connection retries
+				_, cloudConn, err := r.cloudConnSvc.AcquireConnection(ctx)
+				select {
+				case connectionChan <- packages.DeferredConnectionResponse{
+					Client: packagespb.NewPackageServiceClient(cloudConn),
+					Err:    err,
+				}:
```

Comment:
readability -- took me a second to understand that the connectionChan write doesn't fallthrough and return, it moves to next loop iteration. this is probably a me-and-go issue, not an actual readability issue